commands/addons/rename.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+let cli = require('heroku-cli-util');
+let co  = require('co');
+
+let run = cli.command(function (ctx, api) {
+    return co(function* () {
+        let addon    = yield api.get(`/addons/${ctx.args.addon}`);
+        let addonUrl = `/apps/${addon.app.id}/addons/${ctx.args.addon}`;
+
+        yield api.patch(addonUrl, {name: ctx.args.name});
+
+        let oldName = ctx.args.addon;
+        let newName = cli.color.magenta(ctx.args.name);
+
+        cli.log(`${oldName} successfully renamed to ${newName}.`);
+    });
+});
+
+let topic = 'addons';
+
+module.exports = {
+    topic:       topic,
+    command:     'rename',
+    wantsApp:    false,
```

Comment:
We might want this to be true so that you can do `heroku addons:rename -a app DATABASE my-new-name`